diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt

```diff
@@ -51,6 +54,11 @@ import org.jetbrains.kotlin.psi.KtTryExpression
  * - braces around `else`/`catch`/`finally`/`while` (in `do-while` loop)
  */
 class BlockStructureBraces(private val configRules: List<RulesConfig>) : Rule("block-structure") {
```

Comment:
Are you moving comments when the brace is not yet added, or are you moving them after? Because in the latter case it means that for some time the AST is not entirely valid. Is brace added in this rule too?

Reply:
This rule checks spaces near braces, if there aren't braces - this rule won't work.

Reply:
Yes, but the issue originated from the case where the braces have been by the rule (another rule?). Does that mean that after braces are added and until comments are moved, the tree is in invalid state?

Reply:
Right, so the issue is not because `BracesInConditionalsAndLoopsRule` inserts a brace in a wrong place, but because `BlockStructureBraces` moves it incorrectly. OK then

Reply:
`BracesInConditionalsAndLoopsRule` correct put braces. `BlockStructure Braces` incorrect fix spaces between braces and comment